### PR TITLE
FE-339  Fetch campaigns from API

### PR DESCRIPTION
**Pull request by jbianciotto**
# Description

Removed the campaigns json mock and retrieve them from backend API

# Jira tickets

https://trialtech.atlassian.net/browse/FE-339

# Screenshots

If you think a screenshot would be helpful to show this change, please attach it here :grin:

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue) :bug:
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) :bomb::boom:
- [ ] Configuration / Setup :wrench:
- [ ] Clean up / Technical debt :hammer:

## Apps affected by this change

- [x] Web
- [ ] Landing
- [ ] Management portal

## Project code modified by this change

- [x] Web
- [ ] Landing
- [ ] Management portal
- [ ] Packages

# Checklist:

- [x] I have performed a self-review of my own code :innocent:
- [x] I have tested my change in both resolutions, desktop :computer: and mobile :iphone:
- [x] My code is clean and it follows the guidelines of this project :nail_care:
- [x] My changes don't generate new warnings :ok_hand:

### CODE REVIEW

File: apps/web/app/[lang]/components/campaigns/Campaign.tsx [hunk 1]
```diff
@@ -13,26 +13,26 @@ export const Campaign: React.FC<CampaignProps> = ({
...
         />
-        <H3 className="!mt-4 !pb-2 !text-2xl !leading-7">
-          <span className="line-clamp-2">{title}</span>
-        </H3>
-        <Link href={link}>
-          <L1 label={`${t('seeMore')} →`} />
-        </Link>
       </div>
+      <Tag
+        text={tag || String.fromCharCode(160)} //160 = &nbsp;
```
Comment (libattistella) [inline apps/web/app/[lang]/components/campaigns/Campaign.tsx:27]: Why this?

File: apps/web/api/campaigns/requests.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  if (data.total_pages > 1) {
+    console.warn('More than 1 page was returned.');
+  }
+  const campaignsResponse = data.page;
+  const campaigns: Campaign[] = campaignsResponse.map((campaign) => {
+    return {
+      condition: campaign.conditions.length > 0 ? campaign.conditions[0] : null,
+      brief_title: campaign.brief_title,
+      image: campaign.promotional_image,
+      link: `${process.env.NEXT_PUBLIC_DOMAIN}/ensayo-clinico/${campaign.id}`,
```
Comment (libattistella) [inline apps/web/api/campaigns/requests.ts:24]: I think we can star using the new url (studies/...), in some point we will stop using a middleware redirection

File: apps/web/api/campaigns/requests.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export const getCampaigns = async (lang: string) => {
+  const res = await fetch(`${ENDPOINTS.STUDIES_INDEX}?limit=50&lang=${lang}`, {
+    next: {
+      revalidate: process.env.NEXT_PUBLIC_FETCH_CACHE
+        ? REVALIDATE._1D
+        : REVALIDATE.NO_STORE,
+    },
+  });
+  const data: CampaignsApiResponse = await res.json();
+  if (data.total_pages > 1) {
```
Comment (libattistella) [inline apps/web/api/campaigns/requests.ts:15]: Remove for now please

**Review by libattistella — CHANGES_REQUESTED** [review]

File: apps/web/api/campaigns/requests.ts (hunk 3 above)
Comment (jbianciotto) [reply apps/web/api/campaigns/requests.ts:15]: Wouldnt it be nice to have someway to detect when we are returning more results and need adding the next pages consumption?

File: apps/web/app/[lang]/components/campaigns/Campaign.tsx (hunk 1 above)
Comment (jbianciotto) [reply apps/web/app/[lang]/components/campaigns/Campaign.tsx:27]: If tag is empty campaigns wont show up aligned, that is why I added the blank space to reserve the space and dont break the layout

File: apps/web/api/campaigns/requests.ts (hunk 2 above)
Comment (jbianciotto) [reply apps/web/api/campaigns/requests.ts:24]: I dont understand. Instead of /ensayo-clinico/ it should be /studies/ ?

Comment (libattistella) [reply apps/web/api/campaigns/requests.ts:24]: Yep, `/ensayo-clinico/` is part of an url that belongs to the old site. As you may know, we don't have that path segment as part of our studies in this project. For compatibility with the old site, and the campaigns urls going around social networks, etc, we have added a middleware redirection. But the real url for our futures campaign will be `/studies/[id]?campaign_id=[cid]`

File: apps/web/api/campaigns/requests.ts (hunk 3 above)
Comment (libattistella) [reply apps/web/api/campaigns/requests.ts:15]: Is not something that we will be happening in the near future, neither an error, and we will add a pagination mechanism here in some point. Is part of tech debt

File: apps/web/app/[lang]/components/campaigns/Campaign.tsx (hunk 1 above)
Comment (libattistella) [reply apps/web/app/[lang]/components/campaigns/Campaign.tsx:27]: Ok, please add a screenshot to see what you mean. Are you planning handle the case where a tag is longer and takes two lines?

File: apps/web/api/campaigns/requests.ts (hunk 2 above)
Comment (jbianciotto) [reply apps/web/api/campaigns/requests.ts]: endpoint will be /studies/[id], we dont have campaing_id in the data response

File: apps/web/api/endpoints.ts [hunk 4]
```diff
@@ -14,4 +14,5 @@ export const ENDPOINTS = {
     USERS: `${BLOG_BASE_URL}/users`,
   },
   STUDIES: `${BASE_URL}/studies`,
+  STUDIES_INDEX: `${BASE_URL}/studies/reduced_index`,
```
Comment (libattistella) [inline apps/web/api/endpoints.ts:17]:
```suggestion
  ACTIVE_STUDIES: `${BASE_URL}/studies/reduced_index`,
```

**Review by libattistella — CHANGES_REQUESTED** [review]

File: apps/web/app/[lang]/components/campaigns/Campaigns.types.ts [hunk 5]
```diff
@@ -8,11 +8,13 @@ export type Campaign = {
 export type CampaignsProps = {
   campaigns: Campaign[];
   perPage?: number;
+  locale: string;
 };
 
 export type CampaignProps = {
```
Comment (libattistella) [inline apps/web/app/[lang]/components/campaigns/Campaigns.types.ts:14]: export type CampaignProps = LolacaProps & { ... }

File: apps/web/app/[lang]/components/campaigns/Campaigns.types.ts [hunk 6]
```diff
@@ -8,11 +8,13 @@ export type Campaign = {
...
   tag: string;
   title: string;
   link: string;
   image: string;
+  locale: string;
```
Comment (libattistella) [inline apps/web/app/[lang]/components/campaigns/Campaigns.types.ts:19]: Use `LocaleProps` type

File: apps/web/app/[lang]/components/campaigns/Campaigns.types.ts [hunk 7]
```diff
@@ -8,11 +8,13 @@ export type Campaign = {
 export type CampaignsProps = {
   campaigns: Campaign[];
   perPage?: number;
+  locale: string;
```
Comment (libattistella) [inline apps/web/app/[lang]/components/campaigns/Campaigns.types.ts:11]: Use `LocaleProps` type

**Review by libattistella — CHANGES_REQUESTED** [review]

**Review by libattistella — APPROVED** [review]

**Pull request merged**